lib/cocoapods-downloader/git.rb

```diff
@@ -52,7 +52,7 @@ def self.preprocess_options(options)
       #
       def self.commit_from_ls_remote(output, branch_name)
         return nil if branch_name.nil?
-        encoded_branch_name = branch_name.force_encoding(Encoding::ASCII_8BIT)
+        encoded_branch_name = branch_name.encode(Encoding::ASCII_8BIT)
```

Comment:
let me see what I can do because I think we do not want to change the underlying encoding of this string.